### PR TITLE
Create estimate API : - Business and data validation

### PR DESCRIPTION
**Pull request by pintu-eGov**

### CODE REVIEW

File: backend/estimate-service/src/main/java/org/egov/works/producer/Producer.java [hunk 1]
```diff
@@ -3,9 +3,9 @@
 import lombok.extern.slf4j.Slf4j;
 import org.egov.tracer.kafka.CustomKafkaTemplate;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-@Component
+@Service
```
[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/java/org/egov/works/producer/Producer.java:8]: Copy paste issue.. change it to component.

File: backend/estimate-service/src/main/resources/application.properties [hunk 2]
```diff
@@ -38,8 +38,10 @@ egov.mdms.search.endpoint=/egov-mdms-service/v1/_search
 #----------------Idgen Config---------------------#
 egov.idgen.host=https://works-dev.digit.org
 egov.idgen.path=/egov-idgen/id/_generate
-#egov.idgen.es.serviceRequestId.name=pgr.servicerequestid
-#egov.idgen.es.serviceRequestId.format=PB-PGR-[cy:yyyy-MM-dd]-[SEQ_EG_PGR_ID]
+egov.idgen.estimate.number.name=estimate.number
+egov.idgen.estimate.number.format=EP/[fy:yyyy-yy]/[cy:MM]/SEQ_ESTIMATE_NUM_[d{4}]
```
[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/resources/application.properties:42]: I have kept it here just for record. In either cases, we have to configure in MDMS.

File: backend/estimate-service/src/main/resources/application.properties [hunk 3]
```diff
@@ -38,8 +38,10 @@ egov.mdms.search.endpoint=/egov-mdms-service/v1/_search
...
+egov.idgen.sub.estimate.number.name=sub.estimate.number
```
[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/resources/application.properties:43]: done.

[the remark this answers is not in the document]
Comment (pintu-eGov) [reply backend/estimate-service/src/main/resources/application.properties:43]: I am keeping it like this  "egov.idgen.estimateDetail.number.name=estimateDetail.number". is it fine?

[the remark this answers is not in the document]
Comment (rushang7-eGov) [reply backend/estimate-service/src/main/resources/application.properties:43]: Use estimate.detail
Lowercase and dot notation is used in this file.

File: backend/estimate-service/src/main/resources/application.properties [hunk 4]
```diff
@@ -40,8 +40,8 @@ egov.idgen.host=https://works-dev.digit.org
 egov.idgen.path=/egov-idgen/id/_generate
 egov.idgen.estimate.number.name=estimate.number
 egov.idgen.estimate.number.format=EP/[fy:yyyy-yy]/[cy:MM]/SEQ_ESTIMATE_NUM_[d{4}]
-egov.idgen.sub.estimate.number.name=sub.estimate.number
-egov.idgen.sub.estimate.number.format=EP/[fy:yyyy-yy]/[cy:MM]/SEQ_ESTIMATE_NUM_[d{4}]/SEQ_SUB_ESTIMATE_NUM
+egov.idgen.estimate.detail.number.name=estimateDetail.number
```
Comment (rushang7-eGov) [inline backend/estimate-service/src/main/resources/application.properties:43]: The value of `idname` is also not in camel case but lowercase with dot separator. Follow the pattern.

**Review by rushang7-eGov — APPROVED** [review]

**Pull request merged**